cmd/load-smoke-data/load-smoke-data.go

```diff
@@ -137,7 +138,22 @@ func main() {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	client, err := mongo.Connect(ctx, options.Client().ApplyURI("mongodb://localhost:27017").SetConnectTimeout(5*time.Second))
+	clientOptions := options.Client().ApplyURI("mongodb://localhost:27017").SetConnectTimeout(5 * time.Second)
+	envAuth := os.Getenv(evergreen.MongodbAuthFile)
+	if envAuth != "" {
+		ymlUser, ymlPwd, err := evergreen.GetAuthFromYAML(envAuth)
+		if err != nil {
+			grip.EmergencyFatal(errors.Wrapf(err, "problem getting auth info from %s", envAuth))
+		}
+		if ymlUser != "" {
```

Comment:
Should we error if `ymlUser` is empty? It looks like we could get an empty `ymlUser` from `GetAuthFromYAML`, and then call `mongo.Connect` below. Or is the idea that we would support passing an empty file?

Reply:
The strategy for now is to try to load creds from a yaml auth file, then try and connect without auth if the file is empty or there is an error. I cleaned this up; it should just be Error here instead of EmergencyFatal.